src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py

```diff
@@ -863,50 +859,51 @@ def body(self):
-
-    def __init__(self, test_method):
-        # The resources for this test did not exist so the commands will return 404 errors.
-        # So this test is for the command execution itself.
-        super(NetworkVpnConnectionScenarioTest, self).__init__(__file__, test_method)
-        self.resource_group = 'cli_test1'
-        self.placeholder_value = 'none_existent'
-
-    def test_network_vpn_connection(self):
-        self.execute()
-
-    def body(self):
-        rg = self.resource_group
-        pv = self.placeholder_value
-        allowed_exceptions = "The Resource 'Microsoft.Network/connections/{}' under resource group '{}' was not found.".format(pv, rg)
-        self.cmd('network vpn-connection list --resource-group {0}'.format(rg))
-        self.cmd('network vpn-connection show --resource-group {0} --name {1}'.format(rg, pv),
-            allowed_exceptions=allowed_exceptions)
-        self.cmd('network vpn-connection shared-key show --resource-group {0} --name {1}'.format(rg, pv),
-            allowed_exceptions=allowed_exceptions)
-        self.cmd('network vpn-connection delete --resource-group {0} --name {1}'.format(rg, pv))
-        self.cmd('network vpn-connection shared-key reset --resource-group {0} --connection-name {1}'.format(rg, pv),
-            allowed_exceptions=allowed_exceptions)
-        self.cmd('network vpn-connection shared-key update --resource-group {0} --connection-name {1} --value {2}'.format(rg, pv, 'S4auzEfwZ6fN'),
-            allowed_exceptions=allowed_exceptions)
+        vnet1_name = 'myvnet1'
+        vnet2_name = 'myvnet2'
+        gateway1_name = 'gateway1'
+        gateway2_name = 'gateway2'
+        ip1_name = 'pubip1'
+        ip2_name = 'pubip2'
+
+        subscription_id = MOCKED_SUBSCRIPTION_ID \
+            if self.playback \
+            else self.cmd('account list --query "[?isDefault].id" -o tsv')
+
+        self.cmd('network public-ip create -n {} -g {}'.format(ip1_name, rg))
+        self.cmd('network public-ip create -n {} -g {}'.format(ip2_name, rg))
+        self.cmd('network vnet create -g {} -n {} --subnet-name GatewaySubnet'.format(rg, vnet1_name))
+        self.cmd('network vnet create -g {} -n {} --subnet-name GatewaySubnet --subnet-prefix 10.0.1.0/24'.format(rg, vnet2_name))
+
+        subnet1_id = '/subscriptions/{subscription_id}/resourceGroups' \
+            '/{rg}/providers/Microsoft.Network/virtualNetworks/{vnet1}/subnets/GatewaySubnet' \
+            .format(subscription_id=subscription_id, rg=rg, vnet1=vnet1_name)
+        subnet2_id = '/subscriptions/{subscription_id}/resourceGroups' \
+            '/{rg}/providers/Microsoft.Network/virtualNetworks/{vnet2}/subnets/GatewaySubnet' \
+            .format(subscription_id=subscription_id, rg=rg, vnet2=vnet2_name)
+
+        self.cmd('network vpn-gateway create -g {0} -n {1} --subnet-id {2} --public-ip-address {3}'
+                 .format(rg, gateway1_name, subnet1_id, ip1_name))
```

Comment:
the command will fail if anything goes wrong here, if it didn't, the following command will fail when it tries to connect the two gateways.  If the second command passes but didn't create properly, then we have a bigger problem on our hands :)

---
In reply to: [74603966](https://github.com/Azure/azure-cli/pull/687#discussion_r74603966) [](ancestors = 74603966)